vmdb/app/models/vim_performance_state.rb

```diff
@@ -191,4 +194,13 @@ def self.capture_numvcpus(obj)
     # A larger patch should be done outside of a z-stream release
     return obj.hardware.logical_cpus
   end
+
+  def self.capture_logical_cpus(obj)
+    # FIXME: this should probably include capture_numvcpus
+    if obj.respond_to?(:hardware) && obj.hardware
+      return obj.hardware.logical_cpus
+    elsif obj.respond_to?(:container_node) && obj.container_node.hardware
```

Comment:
I think this logic should just be moved up to the poorly named `capture_numvcpus` method.  Also, the `case` statement in `metric/processing` will have to be updated to not restrict to only `VmOrTemplate`.

There should be a separate PR to rename `numvcpus` to `logicalcpus` throughout all the metrics and reporting.